sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataTypeExpression.scala

```diff
@@ -65,3 +65,14 @@ object IntegralTypeExpression {
     e.dataType.isInstanceOf[IntegralType]
   }
 }
+
+object DecimalExtractor {
```

Comment:
If the below `DecimalExpression` not used at all. Shall we rename this to `DecimalExpression`?

Reply:
I choose different name just because this unapply is different which does not returns Boolean. It works more as extract ing the parameters.

Reply:
I know that. But I see you have changed it.